.github/workflows/test.yml

```diff
@@ -335,25 +336,63 @@ jobs:
         with:
           name: verdaccio-log-${{ matrix.template }}
           path: packages/pwa-kit-create-app/local-npm-repo/verdaccio-windows-${{ matrix.template }}.log
-  lighthouse:
+
+  # NOTE: Commenting out this lighthouse test because it's run against a _local_ server that skews the performance metrics to be better than they really are.
+  # TODO: perhaps run lighthouse in another workflow (e2e one) that deploys to MRT and test against that.
+  # lighthouse:
+  #   needs: changelog-check
+  #   strategy:
+  #     fail-fast: false
+  #   runs-on: ubuntu-latest
+  #   env:
+  #     PROJECT_DIR: generated-retail-react-app-test-project
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v4
+
+  #     - name: Setup Ubuntu Machine
+  #       uses: "./.github/actions/setup_ubuntu"
+  #       with:
+  #         node: ${{ env.MRT_NODE }}
+
+  #     - name: Generate retail-react-app-test-project
+  #       run: |-
+  #         node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir ${{ env.PROJECT_DIR }}
+  #       env:
+  #         GENERATOR_PRESET: retail-react-app-test-project
+  #       timeout-minutes: 8
+
+  #     - name: Run Lighthouse CI on the generated project
+  #       uses: "./.github/actions/lighthouse_ci"
+  #       with:
+  #         project_dir: ${{ env.PROJECT_DIR }}
+  #         config_filename: lighthouserc.retail-react-app.js
+
+  bundle-sizes:
     needs: changelog-check
     strategy:
       fail-fast: false
-      matrix:
-        node: [22] # Should match node version supported by MRT.
     runs-on: ubuntu-latest
+    env:
+      PROJECT_DIR: generated-retail-react-app-test-project
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Setup Node
-        uses: actions/setup-node@v4
-        with:
-          node-version: ${{ matrix.node }}
-          cache: npm
-
       - name: Setup Ubuntu Machine
         uses: "./.github/actions/setup_ubuntu"
+        with:
+          node: ${{ env.MRT_NODE }}
+
+      - name: Generate retail-react-app-test-project
+        run: |-
+          node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir ${{ env.PROJECT_DIR }}
+        env:
+          GENERATOR_PRESET: retail-react-app-test-project
+        timeout-minutes: 8
```

Comment:
Did you consider creating a reusable action for generating a project?

Reply:
I did. But there are already existing actions that generate a project (like `generate_app` and `e2e_generate_app`) and they're not what I needed. 

I guess we can create a new action and call it like `generate_app_with_preset`.